src/processout/processout.ts

```diff
@@ -740,7 +740,7 @@ module ProcessOut {
                     var confs = [];
                     for (var conf of data.gateway_configurations) {
                         conf.getInvoiceActionURL = this.buildGetInvoiceActionURL(config.invoiceID, conf);
-                        conf.handleInvoiceAction = this.buildHandleInvoiceAction(config.invoiceID, conf);
+                        conf.handleInvoiceAction = this.buildHandleTokenInvoiceAction(config.invoiceID, conf, config.customerTokenID);
```

Comment:
This is actually the main point of this entire change, where we provide `handleInvoiceAction` (exposed by `hookForInvoice` as well) which can support token or non-token customer action.